### PR TITLE
[176931926 finished] database built for resources, topics, and many-t…

### PR DESCRIPTION
**Pull request by izzygerrard**
Built SQL Database for:
- resources
- topics
- many to many relationships

Ideally, we will be able to make calls from the database to connect the two main pages of the app (the map and the resources page): 
Screen Shot 2021-03-16 at 11 57 12 PM

### CODE REVIEW

File: db/seeds.rb [hunk 1]
```diff
@@ -47,18 +47,37 @@
...
+blm = Topic.create(topic: "BLM")
+lgbtq = Topic.create(topic: "LGBTQ+")
+legal = Topic.create(topic: "Legal")
+physical_health = Topic.create(topic: "Physical Health")
+mental_wellbeing = Topic.create(topic: "Mental Wellbeing")
+immediate_danger = Topic.create(topic: "Immediate Danger")
+domestic_issues = Topic.create(topic: "Domestic Issues")
+general = Topic.create(topic: "General")
+
+#topics = ["Legal", "Physical Health", "Mental Wellbeing", "Immediate Danger", "Domestic Issues", "General"]
```
Comment (robertqiu110) [inline db/seeds.rb:59]: You can probably remove these comments since you have a better method of doing the same thing now.

File: db/seeds.rb [hunk 2]
```diff
@@ -47,18 +47,37 @@
...
+connector1 = Connector.new(
+    #resource_id_id: resource1.id,
+    #topic_id_id: topic2.id
+    resource: resource1,
+    #resource_id_id: resource1.id,
+    topic: legal
 )
+connector1.save!
+
+#lgbtq = Topic.where(topic: "LGBTQ+")
```
Comment (robertqiu110) [inline db/seeds.rb:76]: Same thing here, these comments are outdated and can be removed.

File: app/models/connector.rb [hunk 3]
```diff
@@ -2,15 +2,11 @@
...
 #
-#  id             :integer          not null, primary key
-#  resource_id_id :integer
-#  topic_id_id    :integer
-#
-# Indexes
-#
-#  index_connectors_on_resource_id_id  (resource_id_id)
-#  index_connectors_on_topic_id_id     (topic_id_id)
+#  id          :integer          not null, primary key
```
Comment (robertqiu110) [inline app/models/connector.rb:5]: I like how you define the structure of the table in comments at the top of the file, it really helps in understanding the structure and the codebase quickly

File: db/seeds.rb [hunk 4]
```diff
@@ -47,18 +47,37 @@
...
+#topics.each do |name, t|
+#Topic.create(topic: t)
+#end
 
-topic2 = Topic.new(
-    topic: "LGBTQ+"
+#one way to do it
+Connector.destroy_all
+
+connector1 = Connector.new(
```
Comment (robertqiu110) [inline db/seeds.rb:67]: Not sure if you still need the comments in here, but if you get rid of them you can just condense this into one line.

**Review by robertqiu110 — COMMENTED** [review]
**What is the goal of this PR?**
The goal of this PR is to simplify the database model by reducing extraneous fields just down to 3: id, resource_id, and topic_id.

**Were you able to view screenshots, story cards, and or test the app?**
Yes, there was a screenshot attached to the PR.

**Is the PR in a shape that makes it easy to review?**
Yes, everything is pretty concise and the author left a comment explaining what they did and a screenshot. I wish they added a bit more detail to the comment, although the changes are quite small so it makes sense.

**What are the status of the tests?**
I don't think there was any tests in place to check against the commits, and looking at their test coverage it is pretty low at 18%. So maybe they could write more tests and start running tests against every commit to ensure nothing merged in to master will cause tests to break.

**What is the style of the code like?**
The code has good style, naming conventions and spacing all seem to be good and consistent. Had a few extraneous comments though.

**Is there anything missing from this PR that should be included?**
Not really.

**Feature Design: Are there any parts of this feature that are cause for concern?**
Nope, I don't think so. It seems to good to me.

---

### Overall Assessment

**What does this PR do well?**
It is very concise and easy to look at, and does what it is intended. Style is good and everything is consistent.

**What is something that could be improved with this PR?**
Could've gotten rid of a few comments of old code, as well as provided more information in the author comment. These are all small complaints though, overall it was pretty well done.

File: app/models/connector.rb (hunk 3 above)
Comment (wwang3990) [reply app/models/connector.rb:5]: Agreed, I think these comments help new viewers understand the structure of the table quickly and can also serve as a refresher for people who haven't viewed the file in a while.

File: db/seeds.rb [hunk 5]
```diff
@@ -47,18 +47,37 @@
...
-    topic: "BLM"
-)
+blm = Topic.create(topic: "BLM")
+lgbtq = Topic.create(topic: "LGBTQ+")
+legal = Topic.create(topic: "Legal")
+physical_health = Topic.create(topic: "Physical Health")
+mental_wellbeing = Topic.create(topic: "Mental Wellbeing")
+immediate_danger = Topic.create(topic: "Immediate Danger")
+domestic_issues = Topic.create(topic: "Domestic Issues")
+general = Topic.create(topic: "General")
```
Comment (wwang3990) [inline db/seeds.rb:57]: This does feel a bit repetitive; so if you plan to create more topics in the future you might want to consider DRY-ing this up.

**Pull request merged**